website/src/app/[lang]/[region]/donate/campaign/[campaign]/page.tsx

```diff
@@ -31,32 +34,85 @@ export default async function Page({ params }: CampaignPageProps) {
+		? await getLatestExchangeRate(firestoreAdmin, campaign.goal_currency)
+		: 1.0;
+
+	const contributions = campaign.contributions ?? 0;
+	const amountCollected = Math.round((campaign.amount_collected_chf ?? 0) * exchangeRate);
+	const percentageCollected = campaign.goal ? Math.round((amountCollected / campaign.goal) * 100) : undefined;
+	const daysLeft = daysUntilTs(campaign.end_date.toDate());
+
+	if (campaign.goal) {
+		console.log((campaign.amount_collected_chf * exchangeRate) / campaign.goal);
+	}
+
 	// TODOs
 	// - styling
-	// - add name
-	// - add total amount
-	// - add goal and progress bar if set
+	// - wording
 	return (
 		<BaseContainer className="mx-auto flex max-w-3xl flex-col pt-8 md:pt-16">
-			<div className="flex flex-col items-center">
+			<div className="mb-14 flex flex-col items-center">
 				<Typography size="5xl" weight="bold" color="accent">
 					{campaign.title}
 				</Typography>
+				<Typography size="xl" weight="bold" color="accent" className="mt-4">
+					{translator.t('campaign.by', { context: { creator: campaign.creator_name } })}
+				</Typography>
 				<Typography size="xl" weight="medium" className="mt-4">
 					{campaign.description}
 				</Typography>
-				<div className="mt-16 w-full">
-					<OneTimeDonationForm
-						lang={params.lang}
-						region={params.region}
-						translations={{
-							amount: translator.t('amount'),
-							submit: translator.t('button-text'),
-						}}
-						campaignId={params.campaign}
-					/>
+			</div>
+			{percentageCollected != undefined && (
+				<div className="mb-4 w-full">
+					<Progress value={percentageCollected} />
 				</div>
+			)}
+			<div className="mb-14 flex flex-col items-center">
+				<Typography size="2xl" weight="bold" color="accent">
+					{translator?.t(campaign.goal ? 'campaign.collected-goal' : 'campaign.collected', {
+						context: {
+							count: contributions,
+							amount: amountCollected,
+							currency: campaign.goal_currency,
+							total: campaign.goal,
+						},
+					})}
+				</Typography>
 			</div>
+			{daysLeft >= 0 && (
+				<Fragment>
+					<div className="flex flex-col items-center">
+						<Typography size="xl" weight="bold" color="accent">
+							{translator?.t('campaign.days-left', { context: { count: daysLeft } })}
+						</Typography>
+					</div>
+					<div className="mt-4 w-full">
+						<OneTimeDonationForm
+							lang={params.lang}
+							region={params.region}
+							translations={{
+								amount: translator.t('amount'),
+								submit: translator.t('button-text'),
+							}}
+							campaignId={params.campaign}
+						/>
+					</div>
+				</Fragment>
+			)}
+			{daysLeft < 0 && (
+				<div className="flex flex-col items-center">
+					<Typography size="xl" weight="bold" color="accent">
+						{translator?.t('campaign.ended', { context: { count: daysLeft } })}
+					</Typography>
+				</div>
+			)}
 		</BaseContainer>
 	);
 }
+
+const daysUntilTs = (ts: Date): number => {
```

Comment:
I would move this to `/shared/src/utils/date.ts`. That's where other date utility functions are located.